src/scripts/active-inactive-orgunit/index.js

```diff
@@ -0,0 +1,351 @@
+ */
+function OrganisationUnitActivator(conf) {
+    this.endpoints = {
+        DATAELEMENTGROUP: "/dataElementGroups/UID?fields=dataElements[id,attributeValues]",
+        ORGANISATION_UNITS_BY_UID: "/organisationUnits/UID?fields=level",
+        ORGANISATION_UNITS_BY_ORGANISATION_UNIT_GROUP: "/organisationUnitGroups/UID?fields=organisationUnits[id,closedDate,openingDate]",
+        ORGANISATION_UNITS_BY_PARENT_AND_LEVEL: "/organisationUnits/UID?level=LEVELVALUE&fields=id,closedDate,openingDate",
+        DATAVALUE_SETS: "/dataValueSets",
+        DATAVALUES_RESULT: "/dataValueSets?orgUnit=ORGUNIT&dataElementGroup=DATAELEMENTGROUP&startDate=1900-01-01&endDate=3017-07-01"
+    };
+
+    this.attributeUids = {
+        LEVEL: conf.attributes.level,
+        PARENT: conf.attributes.parent,
+        ORGUNITGROUP: conf.attributes.orgUnitGroup,
+        PERIOD: conf.attributes.period
+    };
+
+    this.dataElementGroup = conf.dataElementGroup;
+
+    //common auth
+    this.requestOptions = {
+        headers: {
+            "authorization": basicAuth(conf.api.auth.username, conf.api.auth.password),
+            "accept": "application/json",
+            "content-type": "application/json",
+        },
+        url: conf.api.protocol + "://" + conf.api.url
+    }
+
+    //counts asynchronous calls to the server
+    this.asyncCalls = 0;
+    console.log("\nConfig:\n", JSON.stringify(conf, null, "\t"));
+};
+
+/**
+ * Run script
+ */
+OrganisationUnitActivator.prototype.run = function () {
+    console.log("\nLoading script..."); 
+    //Process the dataElement group 
+    this.processDataElementGroup(); 
+};
+
+/**
+ * Prepare the options to make the server request.
+ * @param endpoint The endpoint with the params included
+ * @return an object with url and auth info for the given endpoint
+ */
+OrganisationUnitActivator.prototype.prepareOptions = function (endpoint) {
+    var options = Object.assign({}, this.requestOptions);
+    options.url += endpoint;
+    return options;
+}
+
+/**
+ * Prepare the data Elements and load the organisationUnits for each dataelement in a given data element group
+ */
+OrganisationUnitActivator.prototype.processDataElementGroup = function () {
+    console.log("\nLoading dataElements...");
+    var _this = this;
+    var endpoint = this.endpoints.DATAELEMENTGROUP.replace("UID", this.dataElementGroup);
+    var url = this.prepareOptions(endpoint);
+    console.info("Request the dataelements from a dataelementgroup ", "URL: " + url.url);
+    this.asyncCalls++
+    request(url, function (error, response, body) {
+        if (error != undefined) {
+            console.error("Server not found " + error);
+            _this.asyncCalls--;
+            return;
+        }
+
+        var dataElements = JSON.parse(body).dataElements;
+        console.info("Found " + dataElements.length + " dataElements \n\t" + dataElements.map(function (dataElement) { return dataElement.id }).join("\n\t"));
+
+        var dataValues = [];  
```

Comment:
If dataValues is just initialised once and reused all over the script. Why don't we add to 'this' so we don't have to pass it as a parameter?